packages/odie-client/src/components/send-message-input/attachment-button.tsx

```diff
@@ -35,31 +32,19 @@ const getPlaceholderAttachmentMessage = ( file: File ) => {
 
 export const AttachmentButton: React.FC< {
 	attachmentButtonRef?: React.RefObject< HTMLElement >;
-} > = ( { attachmentButtonRef } ) => {
+	externalFile?: File | null;
+	inferredClientId?: string;
+} > = ( { attachmentButtonRef, externalFile, inferredClientId } ) => {
 	const { chat, addMessage, trackEvent, isUserEligibleForPaidSupport } = useOdieAssistantContext();
 	const { data: authData } = useAuthenticateZendeskMessaging(
 		isUserEligibleForPaidSupport,
 		'messenger'
 	);
 	const { isPending: isAttachingFile, mutateAsync: attachFileToConversation } =
 		useAttachFileToConversation();
-	const { zendeskClientId } = useSelect( ( select ) => {
-		const helpCenterSelect: HelpCenterSelect = select( HELP_CENTER_STORE );
-		return {
-			zendeskClientId: helpCenterSelect.getZendeskClientId(),
-		};
-	}, [] );
-
-	const inferredClientId = chat.clientId ? chat.clientId : zendeskClientId;
 	const onFileUpload = useCallback(
-		async ( event: React.ChangeEvent< HTMLInputElement > ) => {
-			if (
-				authData &&
-				chat.conversationId &&
-				inferredClientId &&
-				event.currentTarget.files?.length
-			) {
-				const file = event.currentTarget.files[ 0 ];
+		async ( file: File | undefined ) => {
+			if ( authData && chat.conversationId && inferredClientId && file ) {
```

Comment:
Man, Odie's code is so damn convoluted. Since this is the only place it is used, let's please change `useAttachFileToConversation` to accept whatever params we need and get rid of the `useCallback`. We can then add the following to the mutation.
```js
onSuccess() {
   addMessage( getPlaceholderAttachmentMessage( file ) );
   trackEvent( 'send_message_attachment', { type: file.type } );
}
```